docs/extensions/readme.md

```diff
@@ -1173,3 +1175,56 @@ An object with an optional property.
   }
 }
 ```
+
+## x-ms-header-collection-prefix
+Use `x-ms-header-collection-prefix` to deal with prefixes for headers. Has different behavior if it refers to a request header or a response header:
+
+- Request header: All keys in the request headers will be prefixed with the prefix value before being sent to the service.
+- Response header: Only response headers that start with the prefix specified here will be returned to users. Additionally, the prefix will be stripped from the response header key before being returned to users
+
+Additionally, applying this extension to a schema forces the schema to become a dictionary.
+
+**Schema**: `string`. Name of the prefix you want to append / filter by. A common value for storage libraries is `x-ms-meta-`.
+
+**Request Example**:
+
+```json
+"parameters": [
+  "x-ms-meta": {
+    "type": "string",
+    "x-ms-client-name": "Metadata",
+    "x-ms-header-collection-prefix": "x-ms-meta-"
+  }
+]
+```
+This request parameter will be forced to be a dictionary schema, and all keys in this dictionary will be prefixed with `x-ms-meta-`.
+So, if you input `{"key": "value"}` through this parameter, `{"x-ms-meta-key": "value"}` is what reaches the service.
```

Comment:
>`{"x-ms-meta-key": "value"}`


It would be a header, illustrating it with JSON might be a bit confusing.

Reply:
hey @pakrym does changing it to this clear it up?

`{"key": "value"}` through this parameter, the header `x-ms-meta: {"x-ms-meta-key": "value"}` ...
this is my understanding of what happens to the header param, lmk if this is wrong

Reply:
You can use HTTP request syntax to illustrate
```
GET /path HTTP/1.1
x-ms-meta-key: value
```

Reply:
sweet thanks so much!